src/TransformOperationExecutor.ts

```diff
@@ -158,8 +158,9 @@ export class TransformOperationExecutor {
                 }
 
                 if (!this.isCircular(subValue, level)) {
+                    let transformKey = this.transformationType === "plainToClass" ? newValueKey : key;
```

Comment:
When we transform a plain to a class, then a key has name of raw object. But newValueKey has name of class property.